packages/k6-generator/src/index.js

```diff
@@ -0,0 +1,260 @@
+        uniqueObj = `'${uniqueFile.slice(uniqueFile.lastIndexOf('/') + 1, -5)}'`;
+      }
+      if (!customImport) {
+        customImport = `const SLEEP_DURATION = ${methods['sleep']}`;
+      }
+    }
+  }
+
+  if (!uniqueObj) {
+    uniqueObj = '""';
+  }
+
+  // SPAWN CUSTOM THRESHOLDS AND CHECKS
+  for (let i = 0; i < Object.keys(methods.routes).length; i++) {
+    let methodUrl = Object.keys(methods.routes)[i];
+
+    for (let j = 0; j < methods.routes[methodUrl].length; j++) {
+      const current = methods.routes[methodUrl][j];
+      const { tag } = current;
+      customMetrics = `${customMetrics}
+      "http_req_duration{name:${tag}}": ["p(95)<800"], // threshold on ${tag} requests only
+      "checks{tag:${tag}}": ["rate>0.99"], // rate of successful checks should be higher than 99%`;
+    }
+  }
+
+  // PULL VU OBJECT FROM METHODS FILE - WITHIN DEFAULT FUNCTION
+  const vuObj = Object.assign(methods['vuData']);
+  writeVuObj = `let vuObj = ${JSON.stringify(vuObj)};\n\n`;
+
+  // PARSE THROUGH ALL ROUTES AND ADD TO MAIN CONTENT - WITHIN DEFAULT FUNCTION
+  const baseUrl = methods['baseUrl'];
+
+  for (let i = 0; i < Object.keys(methods.routes).length; i++) {
+    let methodUrl = Object.keys(methods.routes)[i];
+    let fullPath;
+
+    for (let j = 0; j < methods.routes[methodUrl].length; j++) {
+      let current = methods.routes[methodUrl][j];
+      const { path, custom, tag, authReq, payload, uniquePayload, propertyReturned, bodyIncludes } =
+        current;
+      const checkSleep = `check(
+          ${tag}Res, {
+            ['${methodUrl} - ${path} returns successful status']:  (r) => r.status === 200 || r.status === 204,
+            ['${methodUrl} - ${path} returns valid body']: (r) =>
+            r.status === 204 ? true : r.json().hasOwnProperty('${propertyReturned}'),
+            ['${methodUrl} - ${path} returns no error codes']: (r) => r.error_code === 0,
+          },
+          { tag: '${tag}' }
+          );
+          reqBody = {}
+          sleep(SLEEP_DURATION);\n\n`;
+      const setup = `payload = ${JSON.stringify(payload)}
+        if (${payload && payload.length > 0}) {
+            parsePayload(${uniquePayload})
+            }
+        payload = [];
+        tagName = '${tag}Res';`;
+
+      if (custom) {
+        const prop = path.slice(path.indexOf('$') + 2, path.indexOf('}'));
+        if (vuObj[prop]) {
+          fullPath = `${baseUrl}${path.slice(
+            0,
+            path.indexOf('$')
+          )}\$\{vuObj['${prop}']\}${path.slice(path.indexOf('}') + 1)}`;
+        } else {
+          fullPath = `${baseUrl}${path.slice(
+            0,
+            path.indexOf('$')
+          )}\$\{vuObj["${prop}"]\}${path.slice(path.indexOf('}') + 1)}`;
+        }
+      } else {
+        fullPath = `${baseUrl}${path}`;
+      }
+
+      switch (methodUrl) {
+        case 'createUser':
+        case 'login':
+        case 'logout':
+        case 'post':
```

Comment:
If you want to get fancy, a lot of this can probably be put into a method template file then find and replace similar to what you are doing with the base k6 one.